wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -0,0 +1,230 @@
+from django.db import connection
+from django.db.migrations.loader import MigrationLoader
+from django.db.migrations import Migration
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.db.migrations.writer import MigrationWriter
+from django.apps import apps
+from wagtail.blocks import StreamBlock, StructBlock
+
+from wagtail_streamfield_migration_toolkit.migrate_operation import MigrateStreamData
+from wagtail_streamfield_migration_toolkit.operations import (
+    RemoveStreamChildrenOperation,
+    RenameStreamChildrenOperation,
+    RemoveStructChildrenOperation,
+    RenameStructChildrenOperation,
+)
+
+
+class Command(BaseCommand):
+
+    help = "Generate data migrations with the given operations. Note that you can only \
+        generate changes for models in a single app at a time."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--name", help="Use this name for migration file")
+        subparsers = parser.add_subparsers(dest="operation_type")
+
+        rename_parser = subparsers.add_parser("rename", help="rename --help")
+        rename_parser.add_argument("app_label", help="app name")
+        rename_parser.add_argument("old_name", help="name of the block to be renamed")
+        rename_parser.add_argument("new_name", help="new name")
+        rename_parser.add_argument(
+            "paths",
+            nargs="+",
+            action="extend",
+            help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
+        )
+
+        remove_parser = subparsers.add_parser("remove", help="remove --help")
+        remove_parser.add_argument("app_label", help="app name")
+        remove_parser.add_argument("block_name", help="name of the block to remove")
+        remove_parser.add_argument(
+            "paths",
+            nargs="+",
+            action="extend",
+            help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
+        )
+
+        # alter_value_parser = subparsers.add_parser("alter", help="alter --help")
+        # alter_value_parser.add_argument("app_label", help="app name")
+        # alter_value_parser.add_argument("new_value", help="new value")
+        # alter_value_parser.add_argument(
+        #     "paths",
+        #     nargs="+",
+        #     action="extend",
+        #     help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
+        # )
+
+        # TODO support for more operations : TODO alter value
+
+        # TODO determine how to go ahead with multiple blocks AND operations
+
+    def handle(self, *args, **options):
+
+        self.operation_maker = {
+            "rename": self.make_rename_operation,
+            "remove": self.make_remove_operation,
+            # "alter": None,  # TODO
+        }[options["operation_type"]]
```

Comment:
This is correct, but a bit wasteful to create a dict just to index into it once. Given that we only have two options for now, I would suggest doing:

```
if options["operation_type"] == "rename":
    ...
elif ...
```

It could go into a `get_operation_maker` method to keep `handle` clean.

If the number of operations grows greatly, in future you could `getattr(self, f"make_{operation_type}_operation")`.

Would be a nice with Python 3.10's structural pattern matching.